lib/hanami/events/matcher.rb

```
@@ -0,0 +1,24 @@
+module Hanami
+  module Events
+    class Matcher
+      MATCH_ALL_CHARS = '.*'
+      RANGE_PATTERN = '*'
+
+      def initialize(pattern)
+        @pattern = Regexp.new(to_regexp_string(pattern))
+      end
+
+      def match?(event_name)
+        !!@pattern.match(event_name)
+      end
+
+      private
+
+      def to_regexp_string(pattern)
+        pattern.split('.')
+          .map { |p| p == RANGE_PATTERN ? MATCH_ALL_CHARS : p }
```

good point, I think we can raise an error for "invalid" event names 👍